azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -596,10 +596,12 @@ func resourceKubernetesCluster() *schema.Resource {
 							ForceNew: true,
 						},
 						"admin_password": {
-							Type:         schema.TypeString,
-							Optional:     true,
-							Sensitive:    true,
-							ValidateFunc: validation.StringLenBetween(14, 123),
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+							// not doing length validation here otherwise it breaks on existing
+							// clusters with shorter passwords
+							ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
```suggestion
							ValidateFunc: validation.StringLenBetween(8, 123),
```

Hey @feliperoveran, thank you for the PR! Looking good, but could we leave the validation in form of lowering the lower boundary, instead of removing it completely? This way we'll still be able to safeguard from passwords that are too long. I'm not sure whether 8 is low enough, but should be acceptable?

Reply:
8 should be good enough as that used to be the minimum a few months ago :)

Reply:
https://docs.microsoft.com/en-us/rest/api/compute/virtualmachinescalesets/createorupdate#virtualmachinescalesetosprofile
```
Minimum-length (Windows): 8 characters 

Minimum-length (Linux): 6 characters 
```

Reply:
@djsly Thank you so much for finding this. @feliperoveran my apologies, would you mind lowering this to 6?

Reply:
@favoretti  6 is for Linux, this var is for Windows.

Reply:
@djsly Oh, you're completely correct.